include/eve/traits/common_value.hpp

```diff
@@ -8,20 +8,39 @@
 #pragma once
 
 #include <eve/concept/value.hpp>
+#include <eve/concept/simd.hpp>
 #include <eve/traits/as_wide.hpp>
 #include <type_traits>
 
 namespace eve::detail
 {
+  template<typename T>
+  struct find_type_impl {
+    using type = T;
+
+    template<value U>
+    friend consteval find_type_impl<std::conditional_t<
+        scalar_value<T> && scalar_value<U>,
+          decltype(std::declval<T>() + std::declval<U>()),
+          std::conditional_t<
+              (scalar_value<T> && arithmetic_simd_value<U>) ||
+              (arithmetic_simd_value<T> && scalar_value<U>) ||
+              (arithmetic_simd_value<T> && std::same_as<T, U>),
+                std::conditional_t<arithmetic_simd_value<T>, T, U>,
+                void
+          >
+    >> operator%(find_type_impl, find_type_impl<U>) noexcept { return {}; }
+  };
+
   template<typename... Ts>
-  using find_type = decltype(( std::declval<Ts>() + ... ));
+  using find_type = typename decltype((find_type_impl<Ts>{} % ...))::type;
 
   template<typename... Ts>
-  requires(!plain_scalar_value<Ts> || ... )
+  requires(!scalar_value<Ts> || ... )
   auto find_common_value() -> find_type<Ts... >;
 
   template<typename T0, typename... Ts>
-  requires(   (plain_scalar_value<T0> && ... && plain_scalar_value<Ts>)
+  requires(   (scalar_value<T0> && ... && scalar_value<Ts>)
           &&  (std::same_as<T0,Ts> && ...)
           )
   T0 find_common_value();
```

Comment:
This looks very sus btw. Why would this be the case? 
I remember the point of this was - that standard common_type did promotions. We didn't want promotions in simd. Why is first type preferred.

Reply:
We check T0 and all Ts are the same anyway. So we return the first one cause it is the same one than all the other.

Reply:
Ah. Why is this a thing though? What does it do?